### ISSUE TITLE
Memory management (possible leak)

### ISSUE THREAD

**Issue by szamaq** [cmdr2/stable-diffusion-ui#1105]
**Describe the bug**
Creating 2048x2048 images by CPU consumes around 137 GB of RAM. 

1024x1024 + 4x upscaling works on GPU (11.8gb vram). 1536x1536 works only on CPU (not enough vram error) and takes around 16gb of ram WITH operating system(total). 2048x2048 fills whole ram and tries to address another 70+gb. So it looks like server is needed.

**To Reproduce**
car (or any other prompt / negative pr. / samplers)
Seed: 4542298, Sampler: dpmpp_2s_a, Inference Steps: 10, Guidance Scale: 7.5, Model: uberrealisticDreamy_pfg, VAE: vae-ft-mse-840000-ema-pruned, Negative Prompt: deformed, ugly,
Error: [enforce fail at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 137438953472 bytes.

Other settings are defaut. 
**Expected behavior**

It should work and take no more than less than 2x of memory consumed for 1536x1536 which means it should fit in 32gb with OS.

**Screenshots**
https://prnt.sc/Q9tQeOhiGxTI

**Desktop (please complete the following information):**
 - OS: Win11pro up to date
 - Browser: Chrome
 - Version: 111
 - AMD 3700x, 64gb 3200mhz, 3060 12gb, 1tb nvme, python 3.10.6 because auto1111 have issues with anything newer. 

**Additional context**
OS is fresh

**Comment by JeLuF** [comment]
In stable diffusion, the memory usage does not increase linearly with the image size (i.e. total number of pixels). This is because stable diffusion involves computing the Laplacian of the image, which is a second-order partial derivative operator that involves neighboring pixel values. As the image size increases, the number of pixels and the number of neighboring pixels that must be considered also increase, resulting in a higher increase in memory usage.

**Comment by szamaq** [comment]
Thank you in explanation. If generating 1024x1024 takes around 7gb vram (with Windows but without 4x upscaler) so why 4x more pixels (2048x2048) takes 20x more memory instead of 4x?

**Comment by JeLuF** [comment]
Because of the explanation given above. There is a neural network that connects all the pixels with each other. For 4 pixels, the network looks like this:
![image](https://user-images.githubusercontent.com/5852422/229312828-708dc44e-4f76-439b-bc91-62f20dea6ede.png)
The network has 4 nodes and 6 links.

For 9 pixels, it would look like this:
![image](https://user-images.githubusercontent.com/5852422/229312967-2e5cc4f0-ca46-4215-9ac9-b0df2105f1ab.png)
This network has 9 nodes and 36 links.

About twice as many pixels, but already 4.5 times as many links.

The Stable Diffusion model is a little bit more complex, so this is just a simplification.

**Comment by szamaq** [comment]
And this is very good quality explanation!
Thank you good man. It looks like there is no easy way to make high resolution images in SD. I realized that even if I buy old and cheap srv with 256gb ram it wont work because of amount of data to process and the fact that most of them will have broken geometry (5 hands and legs).

Thank you!
